File: 01/go/day1-1.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("../elf_data.txt")
	var all_calories []int
	var cur_calories int = 0
	if err != nil {
		log.Fatalf("failed to open")

	}
	scanner := bufio.NewScanner(file)
	scanner.Split(bufio.ScanLines)

	for scanner.Scan() {
		each_ln := strings.TrimSpace(scanner.Text())
		if len(each_ln) == 0 {
			all_calories = append(all_calories, cur_calories)
			fmt.Printf("🧝‍♀️ number %3d has %5d calories.\n", len(all_calories), cur_calories)
			cur_calories = 0
		} else {
			tmp_cals, _ := strconv.Atoi(each_ln)
			cur_calories += int(tmp_cals)

		}
	}
	file.Close()
	// descending sort, biggest to smallest
	sort.SliceStable(all_calories, func(i, j int) bool {
		return all_calories[i] > all_calories[j]
	})

	fmt.Printf("\nThe 🧝‍♀️ with the most calories has %d calories.\n", all_calories[0])
}

File: 01/python/day1-1.py
#!/usr/bin/env python


max_calories: int = 0
cur_calories: int = 0
elf_count: int = 0

with open("../elf_data.txt") as elf_data:

    for line_calories in elf_data:
        if not line_calories.strip():
            print(f"🧝‍♀️ number {elf_count} had {cur_calories} calories.")
            if cur_calories > max_calories:
                max_calories = cur_calories
            cur_calories = 0
            elf_count += 1
        else:
            cur_calories += int(line_calories)
print()
print(f"The 🧝‍♀️  with the most calories has {max_calories} calories.")
